execute/aggregate.go

```diff
@@ -1,33 +1,159 @@
 package execute
 
 import (
+	"github.com/apache/arrow/go/arrow/memory"
 	"github.com/influxdata/flux"
 	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
 	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute/table"
 	"github.com/influxdata/flux/internal/errors"
-	"github.com/influxdata/flux/memory"
 	"github.com/influxdata/flux/plan"
 )
 
+// AggregateTransformation implements a transformation that aggregates
+// the results from multiple TableView values and then outputs a Table
+// with the same group key.
+//
+// This is similar to NarrowTransformation that it does not modify the group key,
+// but different because it will only output a table when the key is flushed.
+type AggregateTransformation interface {
+	// Aggregate will process the TableView with the state from the previous
```

Comment:
sorry, what is `state` here. the value associated with the given key if it exists?

Reply:
Yes. It's the state associated with the group key that is being aggregated.